### PR TITLE
Update SubmissionConfig test to follow conventions

### PR DESCRIPTION
**Pull request by nawatts**
This updates the SubmissionConfig unit test to follow conventions. Details in inline comments.

### CODE REVIEW

File: src/workflows-app/SubmissionConfig.test.js [hunk 1]
```diff
@@ -1,8 +1,5 @@
-import '@testing-library/jest-dom';
```
Comment (nawatts) [inline src/workflows-app/SubmissionConfig.test.js:1]: `@testing-library` doesn't need to be imported by each test. It's imported in the global test setup.

File: src/workflows-app/SubmissionConfig.test.js [hunk 2]
```diff
@@ -1,8 +1,5 @@
...
-
 import { render, screen, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { act } from 'react-dom/test-utils';
```
Comment (nawatts) [inline src/workflows-app/SubmissionConfig.test.js:5]: `act` rarely needs to be used. `userEvent` methods handle `act` themselves. When it is needed, it should be imported from `@testing-library/react` for consistency.

https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily
https://testing-library.com/docs/react-testing-library/api/#act

File: src/workflows-app/SubmissionConfig.test.js [hunk 3]
```diff
@@ -65,6 +62,8 @@ describe('BaseSubmissionConfig renders workflow details', () => {
 
   it('should render workflow details', async () => {
     // ** ARRANGE **
+    const user = userEvent.setup();
```
Comment (nawatts) [inline src/workflows-app/SubmissionConfig.test.js:65]: > Note that, while directly invoking APIs such as userEvent.click() (which will trigger setup internally) is [still supported in v14](https://testing-library.com/docs/user-event/setup#direct-apis), this option exists to ease the migration from v13 to v14, and for simple tests. We recommend using the methods on the instances returned by userEvent.setup().

https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent

File: src/workflows-app/SubmissionConfig.test.js [hunk 4]
```diff
@@ -138,9 +137,7 @@ describe('BaseSubmissionConfig renders workflow details', () => {
 
     // ** ACT **
     // user clicks on View Workflow Script to open the modal
-    await act(async () => {
-      await userEvent.click(workflowScriptLink);
-    });
+    await user.click(workflowScriptLink);
```
Comment (nawatts) [inline src/workflows-app/SubmissionConfig.test.js:140]: `userEvent` methods do not need to be wrapped in `act`.

**Comment by dillydally414** [comment]
@nawatts Some of these changes will be addressed in https://github.com/DataBiosphere/terra-ui/pull/4063 - I think the only remaining one is fixing the imports, which I could handle in that PR as well if it's easier

**Comment by nawatts** [comment]
Excellent! I'll close this PR then.

**Pull request closed without merging**